### PR TITLE
add support for more languages and upgrade shiki

### PR DESCRIPTION
**Pull request by joshuaflores**
Hey there thanks for making this plugin. I needed syntax highlighting for elixir and noticed that shiki was out of date. I updated that dependency and moved from the now deprecated `oniguruma` to `vscode-oniguruma`.

### CODE REVIEW

**Review by mattbierner — CHANGES_REQUESTED** [review]
The shiki changes look good overall but the PR is doing several things which makes it difficult to review. I suggest keeping this one focused just on the shiki update and handling the language changes in a separate PR

File: src/codeHighlighter.ts [hunk 1]
```diff
@@ -107,11 +108,12 @@ export class CodeHighlighter {
 		}
 
 		if (typeof theme === 'string') {
-			theme = shiki.getTheme(theme as any);
+			// theme = await shiki.getTheme(theme as any);
+			theme = await shiki.loadTheme(currentThemeName)
 		}
 
 		if (theme) {
```
[the remark this answers is not in the document]
Comment (joshuaflores) [reply src/codeHighlighter.ts]: I applied the wrong function here. I reverted but the background color was still showing. For some reason shiki seems to be ignoring `theme.bg`and instead using `colors.editorBackground`. I've applied a patch but perhaps doing an override via css might be better. For future reference the css fix

```
pre.shiki {
   background-color: transparent !important;
}
```

File: src/codeHighlighter.ts [hunk 2]
```diff
@@ -167,12 +169,12 @@ const languages = [
 	{ name: 'lua', language: 'lua', identifiers: ['lua'], source: 'source.lua' },
 	{ name: 'makefile', language: 'makefile', identifiers: ['Makefile', 'makefile', 'GNUmakefile', 'OCamlMakefile'], source: 'source.makefile' },
 	{ name: 'perl', language: 'perl', identifiers: ['perl', 'pl', 'pm', 'pod', 't', 'PL', 'psgi', 'vcl'], source: 'source.perl' },
-	{ name: 'r', language: 'r', identifiers: ['R', 'r', 's', 'S', 'Rprofile'], source: 'source.r' },
+	{ name: 'r', language: 'r', identifiers: ['R', 'r', 's', 'S', 'Rprofile', '\\{\\.r.+?\\}'], source: 'source.r' },
```
[the remark this answers is not in the document]
Comment (joshuaflores) [reply src/codeHighlighter.ts]: I agree that would be a better fix, I needed a quick fix so I just updated the const from 'https://github.com/Microsoft/vscode-markdown-tm-grammar/blob/master/build.js'

What if we just modify `getHighlighter` to something like this? `document.languageId` already returns the language identifier for the file so I'm not sure why the identifier lookup is happening here. By dropping the `getLanguageId` call, I'm getting syntax highlighting on languages not mentioned in `const languages` block

```ts
public async getHighlighter(document: vscode.TextDocument): Promise<(code: string, language: string) => string> {
		const highlighter = await this._highlighter;


		return (code: string, inputLanguage: string): string => {
			const language = inputLanguage || document.languageId;

			if (language && highlighter) {
				try {
					// const languageId = getLanguageId(language);
					if (language) {
						return highlighter.codeToHtml(code, { lang: language });
					}
				} catch (err) {
					// noop
				}
			}

			return code;
		};
	}
```

[the remark this answers is not in the document]
Comment (joshuaflores) [reply src/codeHighlighter.ts]: just created a new pr #45 based on this change. looks good on my end.

File: src/codeHighlighter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuaflores) [reply src/codeHighlighter.ts]: I was wrong on this one: so what I gather is happening is that vscode themes have an optional `settings` section in their declaration file where they can declare styles to be applied to that scope. The problem is that some themes add a setting that with a global scope and that settings applies to everything. It seems that shiki is prioritizing the result of settings over `theme.bg`. In any case I just pushed a fix that should solve this. Although this works this the css fix is the css patch seems more reliable in the long run.

**Comment by mattbierner** [comment]
Closing this for now since it's gotten out of date. I believe the Shiki part has now been tackled but please submit another PR with the language changes if those are still needed

**Pull request closed without merging**